### DISCUSSION TITLE
What is boundary conditions provider operator and how to use it

### DISCUSSION THREAD

**Discussion by kmahajan-cadfem** [ansys/pydpf-core#542] [Q&A] [answered]
I have a simple question,
what is boundary conditions provider operator and how to use it or where can i use it

code
```python
from ansys.dpf import core as dpf
from ansys.dpf.core import operators as ops
from ansys.dpf.core import examples
fp = examples.download_hemisphere()
model = dpf.Model(fp)
ds = dpf.DataSources(model)
st_op = ops.metadata.streams_provider() # operator instantiation
st_op.inputs.data_sources.connect(ds)
my_streams_container = st_op.outputs.streams_container
op = dpf.operators.metadata.boundary_condition_provider() # operator instantiation
op.inputs.data_sources.connect(ds)
print(op.outputs.results_info)
```

output is 

```output
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
~\AppData\Local\Temp\ipykernel_3988\2261196826.py in <module>
     10 op = dpf.operators.metadata.boundary_condition_provider() # operator instantiation
     11 op.inputs.data_sources.connect(ds)
---> 12 print(op.outputs.results_info)

AttributeError: 'Outputs' object has no attribute 'results_info'


```

**Comment by kmahajan-cadfem** [comment]
can any one shed light on this operator? thank you

**Comment by PProfizi** [comment, marked as answer]
Hi @kmahajan-cadfem,

Sorry for the delay. Your question regarding the operator `boundary_condition_provider` requires multiple answers:
- First of all, the reason you get this specific error while trying to use the operator, is due to the fact that the documentation is wrong and the available outputs are actually `results_info_as_field` and `results_info_as_fields_container`. This is a remnant of an old logic used when constructing operator outputs, where if the output type does not exist in the Python API, several outputs are generated based on possible translations to compatible PyDPF-Core types.
- This brings me to the second point, which is that this operator is old, and its use is very targeted and not advised. PyDPF-Core does not expose boundary conditions properly as of today, and this operator is a way to retrieve what information there is on boundary conditions in an RST file and is being formatted for a very specific use. I myself think this operator should be private as it is not ready for public exposure, but due to our policy on retro-compatibility we do not want to remove it from the documentation. What I propose is to edit its description to clarify all of this.
- The third point is that I actually found an issue in its current Python exposition, which makes it unusable as-is. Again, I will look into fixing this or adding a warning to its documentation.

TLDR; this operator is for retrieving the `BC` information in a `.rst` file, yet it is poorly documented and it seems barely working today. Our goal in the future is to properly handle Boundary Conditions as proper entities in PyDPF-Core. I will look into improving its documentation to clarify its use but I would advise to not use it.

**Reply by kmahajan-cadfem** [reply]
@PProfizi, thank you for explaining it in details, it would be great to have documentation of fully working operators only (i.e. only that are tested through CI/CD)
